### ISSUE TITLE
[horovod-tensorflow] Sparse gradient tensor have not aggregated locally

### ISSUE THREAD

**Issue by ustczyb** [horovod/horovod#3328]
Hi, when I trouble-shooting a problem during training via horovod and tensorflow, I found the embedding tensor's gradient, which is a sparse tensor(IndexedSlices), have not aggregated locally before all_reduce or all_gather. That means a embedding tensor may have multiple gradient tensors in a single batch.  It woule cause large network traffic if the batch size is big. I am wondering why not aggregated locally before network communication while treating sparse gradient tensor.

I know adding `sparse_as_dense` option can avoid this. However, when the embedding frequency is not balance(which means a little part of embedding will always be used,  the rest embedding appear only a few times. The word embedding in NLP is one of this situation), one batch training maybe just use several embeddings.In this case, if we treat the embedding gradient tensor as a sparse tensor, and aggregate locally before network communication. I think it would be more efficient.
